packages/coin-sol/src/utils/transactionUtil.ts

```diff
@@ -0,0 +1,163 @@
+import * as params from '../config/params';
+import { TransactionOptions, TransactionType } from '../config/types';
+import { addressToHex, encodeLength, isBase58Format, numberToStringHex } from './stringUtil';
+
+import base58 from 'bs58';
+
+const evenHexDigit = (hex: string) => (hex.length % 2 !== 0 ? `0${hex}` : hex);
+
+export function pubKeyToAddress(publicKey: string): string {
+  const pubKeyBuf = Buffer.from(publicKey, 'hex');
+  return base58.encode(pubKeyBuf);
+}
+
+export class RawTransaction {
+  sigCount: string;
+  signature: string;
+  numberRequireSignature: string;
+  numberReadonlySignedAccount: string;
+  numberReadonlyUnSignedAccount: string;
+  keyCount: string;
+  keys: string;
+  recentBlockHash: string;
+  instructionCount: string;
+  programIdIndex: string;
+  keyIndicesCount: string;
+  keyIndices: string;
+  dataLength: string;
+  data: string;
+
+  constructor(rawTx: {
+    signature?: string;
+    numberRequireSignature: string;
+    numberReadonlySignedAccount: string;
+    numberReadonlyUnSignedAccount: string;
+    keyCount: string;
+    keys: string;
+    recentBlockHash: string;
+    instructionCount: string;
+    programIdIndex: string;
+    keyIndicesCount: string;
+    keyIndices: string;
+    dataLength: string;
+    data: string;
+  }) {
+    this.sigCount = '01';
+    this.signature = rawTx.signature || '';
+    this.numberRequireSignature = rawTx.numberRequireSignature;
+    this.numberReadonlySignedAccount = rawTx.numberReadonlySignedAccount;
+    this.numberReadonlyUnSignedAccount = rawTx.numberReadonlyUnSignedAccount;
+    this.keyCount = rawTx.keyCount;
+    this.keys = rawTx.keys;
+    this.recentBlockHash = rawTx.recentBlockHash;
+    this.instructionCount = rawTx.instructionCount;
+    this.programIdIndex = rawTx.programIdIndex;
+    this.keyIndicesCount = rawTx.keyIndicesCount;
+    this.keyIndices = rawTx.keyIndices;
+    this.dataLength = rawTx.dataLength;
+    this.data = rawTx.data;
+  }
+
+  dataEncode(amount: number | string, decimals?: number | undefined): string {
+    const programIdToNumber = Number(this.programIdIndex);
+    const isNormalTransfer = programIdToNumber === 2;
+    const programIdIndexSpan = isNormalTransfer ? 4 : 1;
+    const dataAlloc = isNormalTransfer ? 12 : 9;
+    const data = Buffer.alloc(dataAlloc);
+
+    data.writeUIntLE(programIdToNumber, 0, programIdIndexSpan);
+
+    if (isNormalTransfer) {
+      const v2e32 = Math.pow(2, 32);
+      const value = Number(amount) * params.LAMPORTS_PER_SOL;
+      const hi32 = Math.floor(value / v2e32);
+      const lo32 = value - hi32 * v2e32;
+
+      data.writeUInt32LE(lo32, programIdIndexSpan);
+      data.writeInt32LE(hi32, programIdIndexSpan + 4);
+    } else {
+      const value = BigInt(
```

Comment:
This SDK cannot contain any native `BigInt`, because this SDK will be used in `react-native` which does not have native `Bigint` support.
Maybe you should try to use `bn.js` instead.

Reply:
Ok, I got it, since react native is not supported `BigInt`, let's me try `bn.js` instead